packager/react-packager/src/DependencyResolver/Package.js

```diff
@@ -92,9 +92,27 @@ class Package {
 }
 
 function getReplacements(pkg) {
-  return pkg['react-native'] == null
-    ? pkg.browser
-    : pkg['react-native'];
+  let rn = pkg['react-native'];
+  let browser = pkg.browser;
+  if (rn == null) {
+    return browser;
```

Comment:
@davidaurelio if rn is null, we don't need to worry about merging rn and browser. But if both are non-null, then line 109 is necessary Also, FYI, i opened the same PR here https://github.com/facebook/node-haste/pull/61 , so we can prob close this one